### DISCUSSION TITLE
[solved] Upgrade Problem

### DISCUSSION THREAD

**Discussion by fbaschin** [netbox-community/netbox#11616] [Help Wanted!] [answered]
After update 3.4.1 to newest version I get

  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/opt/netbox/netbox/netbox/settings.py", line 690, in <module>
    raise ImproperlyConfigured(
django.core.exceptions.ImproperlyConfigured: Unable to import plugin netbox_nagios: Module not found. Check that the plugin module has been installed within the correct Python environment.

**Comment by candlerb** [comment, marked as answer]
When you run `./upgrade.sh` it deletes your old virtualenv and creates a new one.  Therefore, if you had installed some package *by hand* in the old environment, it will be gone.  You simply need to reinstall it.

If you want plugins to be reinstalled automatically on upgrade, add them to a file called `/opt/netbox/local_requirements.txt`
